lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -1072,20 +1149,19 @@ def fit_binding_times(
             The maximum number of iterations to perform. This parameter is forwarded as the `maxiter` argument
             to `scipy.minimize(method="L-BFGS-B")`.
         """
-        self._validate_single_source("Dwelltime analysis")
+        if not self:
```

Comment:
is this a shortcut to `if len(self) < 1` ?

if so, it looks a bit weird and hard to read if someone is not familiar with the codebase

Reply:
Isn't this a pretty common empty check for a list-like class though? I don't mind changing it to `if not len(self):` if you think that is more clear.

But I'd rather not go with the `< 1`. Afaik, length dunders cannot return negative values, so the `< 1` would make me wary whether something special is happening here.